commit 1ac6d9284aa420a1236801eb34c588492f7ab6b1 (refs/changes/60/3160/1)
Author: Jan Wyszynski <jan.wyszynski@joyent.com>
Date:   2018-01-05T01:00:41+00:00 (1 year, 9 months ago)
    
    MANATEE-389 deposed array is missing in stateBackfill
    MANATEE-390 check for deposed array's existence in manatee-adm history

diff --git a/lib/adm.js b/lib/adm.js
index 6e642b0..411ef5d 100644
--- a/lib/adm.js
+++ b/lib/adm.js
@@ -1259,6 +1259,9 @@ function stateBackfill(opts, cb) {
             if (!stat.async) {
                 stat.async = [];
             }
+            if (!stat.deposed) {
+                stat.deposed = [];
+            }
             stat.generation = 0;
             stat.initWal = '0/0000000';
             stat.freeze = {
@@ -1967,8 +1970,8 @@ function annotateHistoryNode(evt, last)
 
     newpeers = {};
     oldpeers = {};
-    nst.deposed.forEach(function (a) { newpeers[a.zoneId] = true; });
-    lst.deposed.forEach(function (a) { oldpeers[a.zoneId] = true; });
+    (nst.deposed || []).forEach(function (a) { newpeers[a.zoneId] = true; });
+    (lst.deposed || []).forEach(function (a) { oldpeers[a.zoneId] = true; });
     for (h in newpeers) {
         if (!oldpeers.hasOwnProperty(h))
             changes.push(sprintf('"%s" deposed',
